harmony/harmony.py

```diff
@@ -364,8 +376,73 @@ def progress(self, job_id: str) -> int:
         can't be reached.
         """
         session = self._session()
-        response = session.get(self._status_url(job_id)).result()
+        response = session.get(self._status_url(job_id))
         if response.ok:
             return int((response.json())['progress'])
         else:
             response.raise_for_status()
+
+
+    def wait_for_processing(self, job_id: str, show_progress=False) -> None:
+        check_interval = 3.0  # in seconds
```

Comment:
Do we want to make these values overridable?

I'm also trying to understand the difference between the check_interval and the sleep interval. Couldn't we just have one interval? It looks like on line 389 intervals is set to 9 seconds and then on 405 we sleep for 0.33 seconds before checking again. I guess I'm a little confused by how long we're sleeping and how often we're calling harmony.

Reply:
`check_interval` is how often we're calling Harmony and is meant to be set to an interval that won't produce undue load on the servers. The sleep interval determines how often we refresh the UI (the optional progress bar gui element). Dividing the two gives us a number we can use to create the cycle'd sequence the loop goes through so that it updates the UI fast but checks the servers slowly.

Reply:
Changing `sleep_interval` to `ui_update_interval`.

Reply:
Even if not overridable, putting it in some sort of constant and moving the various constants to a consistent location (top of the file?) is a good practice.  I think Chris may be responding to a sense of inconsistency and magic in seeing a number but not knowing what it's for or where it came from.  Rule of thumb: if the number isn't 0 or 1, put it in a constant and add a line of docs.  I patiently await you finding 5 places I've personally broken that rule :)

Reply:
(Looking through the code) Also numbers that are obviously well-understood and will basically never ever change, like checking if an HTTP status is 404, you don't need to externalize that.

Reply:
Just FYI that Ryan and I paired to walk through a notebook using this logic and renaming sleep_interval to ui_update_interval made a big difference to grok what the code was doing.

I do agree that pulling out constants is good - defining them here in the function is ok for me too.

Reply:
Good feedback all around. I moved those constants up to the top of harmony.py and commented on their purpose.

Reply:
You can ignore my prior comments, I think what you have is probably alright.